modules/osm/tags.js

```diff
@@ -13,6 +13,33 @@ export function osmSetAreaKeys(value) {
     osmAreaKeys = value;
 }
 
+export const osmLifecyclePrefixes = {
+    // nonexistent, might be built
+    proposed: true, planned: true,
+    // under maintentance or between groundbreaking and opening
+    construction: true,
+    // existent but not functional
+    disused: true,
+    // dilapidated to nonexistent
+    abandoned: true, was: true,
+    // nonexistent, still may appear in imagery
+    dismantled: true, razed: true, demolished: true, destroyed: true, removed: true, obliterated: true,
+    // existent occasionally, e.g. stormwater drainage basin
+    intermittent: true
+};
+
+/** @param {string} key */
+export function osmRemoveLifecyclePrefix(key) {
+    const keySegments = key.split(':');
```

Comment:
`String.prototype.split()` accepts a `limit` parameter; passing in `1` would simplify this function a little.

Reply:
didn't know that :) but I don't quite understand how that would simplify this function

Reply:
I was thinking that you could limit the split to 1 to have the part before the first colon and everything after that first colon. Then you wouldn’t need to deal with indices. But what you have does work.

Reply:
gotcha, sadly JS's `split` function isn't as nice as languages like ruby

 `"a:b:c".split(":", 1)` just returns `["a"]`, not `["a", "b:c"]`